filebeat/input/filestream/internal/input-logfile/harvester_test.go

```diff
@@ -0,0 +1,286 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package input_logfile
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/x-pack/dockerlogbeat/pipelinemock"
+	"github.com/elastic/go-concert/timed"
+	"github.com/elastic/go-concert/unison"
+)
+
+func TestReaderGroup(t *testing.T) {
+	t.Run("assert new group is empty", func(t *testing.T) {
+		rg := newReaderGroup()
+		assert.Equal(t, 0, len(rg.table))
+	})
+
+	t.Run("assert non existent key can be removed", func(t *testing.T) {
+		rg := newReaderGroup()
+		assert.Equal(t, 0, len(rg.table))
+		rg.remove("no such id")
+		assert.Equal(t, 0, len(rg.table))
+	})
+
+	t.Run("assert inserting existing key returns error", func(t *testing.T) {
+		rg := newReaderGroup()
+		ctx, cf, err := rg.newContext("test-id", context.Background())
+		assert.NotNil(t, ctx)
```

Comment:
do not use assert but `require`. Some checks later in the test do require the initial operation to succeed. In order to reduce noise with cascading errors we want to quit the test early. `assert` package does mark the test as failed, but continues to run it. `require` fails immediately.